[Backport 8.6]: Decouple Load tests of Camunda Platform Helm chart

ChrisKujawa opened the pull request:
## Description

Backport of https://github.com/camunda/camunda/pull/40865

> [!Important]
>
> All commits have been easily cherry-picked (as 8.6 ~ 8.7 almost similar)


> * Backport of https://github.com/camunda/camunda/pull/39618
> * Especially the commit https://github.com/camunda/camunda/commit/0352dbe07b5267837a73b39b77f1a17aa24e3376
> * Made adjustments to make it work for 8.7 like:
>     * Adjusting the values file, deploying zeebe and gateway - configure separate
>     * Adjusting load test workflow to reference right service name

<!-- Describe the goal and purpose of this PR. -->


## Related issues

related to https://github.com/camunda/camunda/issues/38829

ChrisKujawa committed:
ci: add new load-test workflow

 * Adds a new load test workflow,
   which will in future replace the old zeebe benchmark workflow
 * The load test workflow decouples the platform and load test chart
   installations
 * To do this, we add an extra values file to configure the platform
   chart, corresponding to the stable version
 * Values are based on previous benchmark chart see
https://github.com/camunda/zeebe-benchmark-helm/blob/49a219f30295fedb047262805d6b49640d143b75/charts/zeebe-benchmark/values.yaml#L1
 * Adjust existing workflows referencing old benchmark workflow and make
   use of load test
 * As this was backported from main, we had to make some adjustments ot
   the paths used, for example the project is still under
   zeebe/benchmark/project

(cherry picked from commit d24b71b67f86de27395816dc2e3e79883b3ab64c)
(cherry picked from commit 79364b8dc6f160caeaf0839d268ba2d1464b3408)
(cherry picked from commit c7a13873583fc6f36848d04bb1f4c06c74b4947a)
(cherry picked from commit f14b49daaa071667f2ce3c434e5f7332ab6ae72f)
(cherry picked from commit 5c4bfac3ef5f2ffb827202be03f0de47e36d4345)
(cherry picked from commit 36a317f88698ad0d5a5cb8a06b4ce8e63a92fa53)
(cherry picked from commit 9dfdae615db752dc9ef16df25a68f4493ef2680b)
(cherry picked from commit 2628a934002fe4948f477a2ca7d0a058e425f362)
(cherry picked from commit 7acd62f5149f76047ba1bc85b491205ad48198bf)
(cherry picked from commit f3b472a27b62608a5f7b1c49dc3858be894ea0f1)
(cherry picked from commit 1c39f986dcb3c9aa97a4cc526151b0ef1ab00173)
(cherry picked from commit 49cfd025671bc1e5a69e5bea904ff3472492b451)
(cherry picked from commit ef73f9eeb35751b45a6ff26b38b9c546f87900b5)

This was restored from commit 79364b8dc6f160caeaf0839d268ba2d1464b3408

ChrisKujawa committed:
ci: prepare load test for 8.7

ChrisKujawa committed:
test: rework 8.7 values

 * Adjust platform values file to work with
   12.x platform chart and properly configure cluster
 * Based on new load tests values from main
https://github.com/camunda/camunda/blob/main/load-tests/camunda-platform-values.yaml
 * Based on older values files from benchmark chart
https://raw.githubusercontent.com/camunda/zeebe-benchmark-helm/refs/tags/zeebe-benchmark-0.3.7/charts/zeebe-benchmark/values.yaml
 * Had to translate changes and configurations to make it work

ChrisKujawa committed:
ci: set correct service name

 * The default service name is camunda, which is the related to the
   new architecture
 * Before we just deployed zeebe and load tested zeebe only
 * Service name has been changed (release name removed)
 * Adjusting here so starter and worker can connect

ChrisKujawa committed:
test: remove unnecessary configs

 * Certain configurations are only necessary for broker or gateway
   we do not need to set them on the other deployments

ChrisKujawa commented on zeebe/benchmarks/camunda-platform-values.yaml:
```diff
@@ -0,0 +1,283 @@
+      zeebe.broker.exporters.elasticsearch.args.index.indexSuffixDatePattern: "yyyy-MM-dd_HH"
+      # Configure a rate limit for all writes, so that we can visualize flow control metrics.
+      zeebe.broker.flowControl.write.enabled: "true"
+      zeebe.broker.flowControl.write.limit: 4000
```
https://github.com/camunda/camunda/pull/40865/commits/ce7fa754b89fa0fd865d123b25e2fd45950b67d9

ChrisKujawa commented on .github/workflows/camunda-load-test.yml:
```diff
@@ -0,0 +1,396 @@
+          --set orchestration.image.tag=${{ needs.calculate-image-tag.outputs.image-tag }}
+          -f https://raw.githubusercontent.com/camunda/camunda/refs/heads/${{ inputs.ref }}/zeebe/benchmarks/camunda-platform-values.yaml
+          ${{ steps.find-chart-release.outputs.chart-release != '' && format('--version {0}', steps.find-chart-release.outputs.chart-release) || '' }}
+          ${{ inputs.stable-vms && '-f zeebe/benchmarks/setup/default/values-stable.yaml' || '' }}
```
@copilot open a new pull request to apply changes based on [this feedback](https://github.com/camunda/camunda/pull/40991#discussion_r2526838106)

ChrisKujawa replied:
[795ad9c](https://github.com/camunda/camunda/pull/40991/commits/795ad9cccf909a0185775a1269cd28b6281f370b)

ChrisKujawa committed:
fix: make use of github.ref

 * We need to make use of the github.ref to resolve the path of the values files
   The given ref is not properly formated, when being a branch or a tag causing
   to fail the deployment for example for release load tests done for tags
 * The triggered workflows will correspond to the right version of the values file so
   this should work

ChrisKujawa committed:
fix: remove duplicate ref input

 * Remove obsolete ref input, as it is confusing to have a workflow revision
   and branch/tag ref
 * Both need to at least have the same parent, as we depend now on the values file on the branch and chart version
   we need to deploy
 * ideally they are aligned or the same - removing the ref makes this easier

ChrisKujawa committed:
fix: change versions to 8.6

 As commits have been backported, we need to adjust the version

ChrisKujawa committed:
ci: bump version of credentials action

ChrisKujawa committed:
docs: fix typo

entangled90 commented on .github/workflows/camunda-load-test.yml:
```diff
@@ -0,0 +1,387 @@
+        with:
+          directory: ./operate/client
+      - uses: ./.github/actions/build-frontend
+        name: Build Tasklist Frontend
```
❓ I was wondering, do we really need to build the frontends?
We do this in all branches actually, so it's not related to this PR

ChrisKujawa replied:
Yeah - sometimes we want to look into the applications. I would like to keep it this way, so we have the actual deliverable tested.

entangled90 reviewed (approved):
Did you test that it works? 

I'll preapprove anyway

ChrisKujawa committed:
ci: remove maven.test.skip=true

 * When the test snapshots are not existing, because of a new release fex
   the building of a docker image with fail
 * This means we can't set up a load test, which is sub-optimal
 * Removing this flag, will build all dependencies

For context https://camunda.slack.com/archives/C013MEVQ4M9/p1763381532908289?thread_ts=1760674300.805349&cid=C013MEVQ4M9
